### PR TITLE
Tenant topics configuration

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This adds a "topics" configuration to Tenant so that it can eventually manage pub/sub event streams with Ensign. We intend to use this to create a subscriber within Tenant to subscribe to the list of topics that Ensign creates/modifies/deletes.

Fixes SC-13883

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Please review the configuration to make sure it's complete and it's understandable how it's intended to be used.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/config/config.go [hunk 1]
```diff
@@ -34,6 +35,7 @@ type Config struct {
 	Auth         AuthConfig          `split_words:"true"`
 	Database     DatabaseConfig      `split_words:"true"`
 	Ensign       EnsignConfig        `split_words:"true"`
+	Topics       SDKConfig           `split_words:"true"`
```
Comment (pdeziel) [inline pkg/tenant/config/config.go:38]: If Tenant needs more event streams in the future, my thinking was that we would just add more of these.

File: pkg/tenant/config/config.go [hunk 2]
```diff
@@ -71,6 +73,12 @@ type EnsignConfig struct {
 	PoolPath string `split_words:"true"`
 }
 
+// Configures an SDK connection to Ensign for pub/sub.
+type SDKConfig struct {
+	Ensign  ensign.Options
+	TopicID string `split_words:"true"`
```
Comment (pdeziel) [inline pkg/tenant/config/config.go:79]: Is this too tied to the SDK?

Comment (bbengfort) [reply pkg/tenant/config/config.go:79]: Does the `ensign.Options` have `envconfig` variables?

Comment (bbengfort) [reply pkg/tenant/config/config.go:79]: I think this is ok for now.

**Review by bbengfort — APPROVED** [review]
Thanks for adding the validation code to the options! It might be a little tightly coupled right now but we're still figuring out the SDK so I think this is the right path forward.

**Pull request merged**